cfgrammar/src/lib/header.rs

```diff
@@ -18,19 +18,18 @@ pub enum HeaderErrorKind {
     ExpectedToken(char),
     DuplicateEntry,
 }
+// Indicates a value prefixed by an optional namespace.
+// `Foo::Bar` with optional `Foo` specified, becomes (Some(("Foo", ...)), ("Bar", ...))
+// Alternately just `Bar` alone without a namespace is represented by (None, ("Bar", ...)).
+pub type Namespaced = (Option<(String, Span)>, (String, Span));
 
 #[derive(Debug, Eq, PartialEq)]
 #[doc(hidden)]
-pub enum Path<'a> {
-    Ident(&'a str, Span),
-    Scoped((&'a str, Span), (&'a str, Span)),
-}
-
-#[derive(Debug, Eq, PartialEq)]
-#[doc(hidden)]
-pub enum Setting<'a> {
-    PathLike(Path<'a>),
-    ArgLike(Path<'a>, Path<'a>),
+pub enum Setting {
+    // A value like `YaccKind::Grmtools`
+    Unitary(Namespaced),
```

Comment:
These names felt maybe a bit different in comparison to `Num`? I was wondering whether `Constructor`, for example, is "key:value" in disguise?